osu.Game/Screens/Play/HUD/HitErrorMeters/ColourHitErrorMeter.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.IEnumerableExtensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Rulesets.Scoring;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Screens.Play.HUD.HitErrorMeters
+{
+    public class ColourHitErrorMeter : HitErrorMeter
+    {
+        private const int max_available_judgements = 20;
+
+        private readonly JudgementFlow judgementsFlow;
+        private readonly BindableList<(Color4 colour, JudgementResult result)> judgements = new BindableList<(Color4 colour, JudgementResult result)>();
```

Comment:
curious on the reason for making a bindable list and a subcomponent. seems like unnecessary complexity.

Reply:
It's easier to maintain judgements capacity like this rather than counting drawables

Reply:
i'm not sure i understand. you're still counting in the end.

Reply:
I tried to make it as you suggested first, but this was not working correctly for high 7K mania difficulties. I tried to remove drawables using `Alpha == 1` for the last one, but it failed since there could be many of them fading out at the same time.
(This is what I was typing and then realized: for that reason I added `FinishTransforms` for each drawables on new judgement arrived)

Reply:
```csharp
DrawableResult
{
    public bool IsRemoved;

    void Remove()
    {
		IsRemoved = true;
        FadeOut(...);
        Expire();
    }
}

Children.Where(c => !c.IsRemoved)
```

is one simple way